# TPOT

We'll now use TPOT to find the best pipeline to our problem

In [40]:
%matplotlib notebook

import numpy as np
from tpot import TPOTClassifier
from sklearn.model_selection import train_test_split



Importing the DATA

In [44]:
X = np.genfromtxt('../features/features_wave_julian_sarah.csv', delimiter=',')
y = np.genfromtxt('../features/output_wave_julian_sarah.csv', delimiter=',')


X_train , X_test, y_train, y_test = train_test_split(X,y,train_size = 0.75 ,test_size = 0.25)

#y_train = y[:int(l*training_ratio)]/2
#y_test = y[int(l*training_ratio):]/2



#y_train = keras.utils.np_utils.to_categorical(y_train.astype(int))
#y_test = keras.utils.np_utils.to_categorical(y_test.astype(int))

In [45]:
tpot = TPOTClassifier(verbosity = 2, n_jobs = -1)

In [46]:
tpot.fit(X_train, y_train)

Generation 1 - Current best internal CV score: 0.86
Generation 2 - Current best internal CV score: 0.86
Generation 3 - Current best internal CV score: 0.86
Generation 4 - Current best internal CV score: 0.86
Generation 5 - Current best internal CV score: 0.86
Generation 6 - Current best internal CV score: 0.86
Generation 7 - Current best internal CV score: 0.8616666666666667
Generation 8 - Current best internal CV score: 0.8616666666666667
Generation 9 - Current best internal CV score: 0.8616666666666667
Generation 10 - Current best internal CV score: 0.8633333333333333
Generation 11 - Current best internal CV score: 0.8633333333333333
Generation 12 - Current best internal CV score: 0.8633333333333335
Generation 13 - Current best internal CV score: 0.8641666666666665
Generation 14 - Current best internal CV score: 0.8641666666666665
Generation 15 - Current best internal CV score: 0.8641666666666665
Generation 16 - Current best internal CV score: 0.8641666666666667
Generation 17 - Curre

TPOTClassifier(config_dict=None, crossover_rate=0.1, cv=5,
        disable_update_check=False, early_stop=None, generations=100,
        max_eval_time_mins=5, max_time_mins=None, memory=None,
        mutation_rate=0.9, n_jobs=-1, offspring_size=None,
        periodic_checkpoint_folder=None, population_size=100,
        random_state=None, scoring=None, subsample=1.0,
        template='RandomTree', use_dask=False, verbosity=2,
        warm_start=False)

Let's check the score of the algorithm found:

In [47]:
tpot.score(X_test, y_test)

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\preprocessing\_function_transformer.py:98: FutureWarning: The default validate=True will be replaced by validate=False in 0.22.
  "validate=False in 0.22.", FutureWarning)


0.8275

In [48]:
tpot.export('tpot_pipeline.py')

# Average CV score on the training set was:0.8275 using PCA then ExtraTreesClassifier


We have already done better using Neural Networks. But in average this is better

Let's try to classify the three players now and see how accurate can we be

In [50]:
X = np.genfromtxt('features_wave_julian_paul_sarah.csv', delimiter=',')
y = np.genfromtxt('output_wave_julian_paul_sarah.csv', delimiter=',')


X_train , X_test, y_train, y_test = train_test_split(X,y,train_size = 0.75 ,test_size = 0.25)

In [51]:
tpot = TPOTClassifier(verbosity = 2, n_jobs = -1)

In [52]:
tpot.fit(X_train, y_train)

Generation 1 - Current best internal CV score: 0.6802481892421357
Generation 2 - Current best internal CV score: 0.6802481892421357
Generation 3 - Current best internal CV score: 0.6802481892421357
Generation 4 - Current best internal CV score: 0.6813701761393208
Generation 5 - Current best internal CV score: 0.6824703768831621
Generation 6 - Current best internal CV score: 0.6824703768831621
Generation 7 - Current best internal CV score: 0.683579949131061
Generation 8 - Current best internal CV score: 0.6836109943995754
Generation 9 - Current best internal CV score: 0.6841711924806647
Generation 10 - Current best internal CV score: 0.6841711924806647
Generation 11 - Current best internal CV score: 0.6863623780796717
Generation 12 - Current best internal CV score: 0.6863623780796717
Generation 13 - Current best internal CV score: 0.6863623780796717
Generation 14 - Current best internal CV score: 0.6863623780796717
Generation 15 - Current best internal CV score: 0.6863732192845496
Gener

TPOTClassifier(config_dict=None, crossover_rate=0.1, cv=5,
        disable_update_check=False, early_stop=None, generations=100,
        max_eval_time_mins=5, max_time_mins=None, memory=None,
        mutation_rate=0.9, n_jobs=-1, offspring_size=None,
        periodic_checkpoint_folder=None, population_size=100,
        random_state=None, scoring=None, subsample=1.0,
        template='RandomTree', use_dask=False, verbosity=2,
        warm_start=False)

In [53]:
tpot.export('tpot3_pipeline.py')

In [54]:
tpot.score(X_test, y_test)

0.666110183639399

# The precision decays significatively for 3 players (0.66). It is hard to differenciate the girl (Sarah) from the two boys (Julien and Paul). We have constated it as well when trying to cluster the data with K-means. When we selected 3 clusters, the algorithm would only split the data in 2 mainly clusters.